src/ds/thread_messaging.h

```diff
@@ -276,6 +293,20 @@ namespace threading
       task.add_task(reinterpret_cast<ThreadMsg*>(msg.release()));
     }
 
+    void add_task(uint16_t tid, std::function<void()>&& f)
+    {
+      Task& task = get_task(tid);
+      task.add_task(new ThreadMsg(std::move(f)));
+    }
+
+    void add_task(std::function<void()>&& f)
+    {
+      add_task(
+        threading::ThreadMessaging::get_execution_thread(
+          threading::MAIN_THREAD_ID),
+        std::move(f));
+    }
```

Comment:
consider something like add_task_on_main as to me an add task without a tid implies you call it on the current thread